lib/puma/client.rb

```diff
@@ -104,6 +104,15 @@ def set_timeout(val)
       @timeout_at = Time.now + val
     end
 
+    # Number of seconds until the timeout elapses.
+    def timeout
+      [@timeout_at - Time.now, 0].max
+    end
+
+    def <=>(other)
```

Comment:
This feels weird to me conceptually. Clients are the same if they time out at the same time?

Reply:
This method was used by `SortedSet` (which calls `#sort!` under the hood to arrange its items), it wasn't meant to convey identity, only ordering. To avoid confusion I changed `@timeouts` back to an Array that calls `#sort_by!` to arrange its items in 51757928a24ae8e3f1980866d1104b79704a758b.